src/compute-client/src/controller.rs

```diff
@@ -408,6 +410,7 @@ where
 pub struct ActiveComputeController<'a, T> {
     compute: &'a mut ComputeController<T>,
     storage: &'a mut dyn StorageController<Timestamp = T>,
+    collections: &'a mut dyn CollectionsController<Timestamp = T>,
```

Comment:
You mean why it's needed by the `ComputeController` here, or in general? I'm assuming the former, and yes, the goal is that its is in fact _not_ needed! It's currently only needed (modulo some cleanups, I think) because the compute controller forwards frontier updates for materialized views to the storage controller. Which won't be needed once the `StorageController` doesn't itself hold Since handles but instead also uses a `CollectionsController`. And the `CollectionsController` already uses persist to learn about upper updates.